### DISCUSSION TITLE
OnRecordBeforeCreateRequest - userid

### DISCUSSION THREAD

**Discussion by IwateKyle** [pocketbase/pocketbase#2174] [Q&A] [answered]
Hi, I have a *messages* collection with a *sender* field where I want to set that value automatically to the logged in userid.

I'm try to do this in the OnRecordBeforeCreateRequest event, but I don't know how to access the current userid for the request.

I'm currently trying to do something like this....
```go

app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
		if e.Record.Collection().Name != "messages" {
			return nil // not a user
		}

		// set default coints value
		e.Record.Set("sender", ****** insert user id here *******)

		return nil
	})
```

**Comment by ganigeorgiev** [comment, marked as answer]
This will be simplified in the near future. For now you can access the authenticated auth record or admin from the request context - https://pocketbase.io/docs/router/#get-logged-admin-or-app-user-in-a-request-hook:

```go
app.OnRecordBeforeCreateRequest("messages").Add(func(e *core.RecordCreateEvent) error {
    // overwrite the submitted sender field for regular users
    authRecord, _ := e.HttpContext.Get(apis.ContextAuthRecordKey).(*models.Record)
    if authRecord != nil {
        e.Record.Set("sender", authRecord.Id)
    }

    return nil
})
```

**Comment by IwateKyle** [comment]
ahh!  Thanks for the fast response.
